ecl/hql/hqlgram.y

```diff
@@ -2697,7 +2697,11 @@ actionStmt
                             if (!recordTypesMatch($3.queryExpr(), $5.queryExpr()))
                                 parser->reportError(ERR_TYPEMISMATCH_RECORD, $4, "Indexes must have the same structure");
                             parser->normalizeExpression($7, type_string, false);
-                            $$.setExpr(createValueFromCommaList(no_keydiff, makeVoidType(), createComma($3.getExpr(), $5.getExpr(), $7.getExpr(), $8.getExpr())));
+
+                            OwnedHqlExpr options = createComma($3.getExpr(), $5.getExpr(), $7.getExpr(), $8.getExpr());
+                            if (!parser->checkAllowed($1, "datafile", "Datafile access"))
```

Comment:
Another reason for moving to a function like this is that this is likely to change in the future.  E.g., checking if the person signing something has right to do that.